### PR TITLE
Fix section mapping, hiding untagged PRs, and hiding untagged issues

### PR DESCRIPTION
**Pull request by tduffield**
So this PR turned into a bit of a beast, but the foundation of it is to get three pieces of functionality working as they are documented. 

The first is the section mapping. I've included some tests to reflect the desired behavior but there are a few key changes. 

1. When `between-tags` is specified, it no longer leaves an empty section header. This was causing duplicate headers to show up when using base. Instead, the first/earliest between-tags tag will be included in the appropriate compare link. 
2. `between-tags` will now aggregate changes between the specified tags (as noted in #450). 
3. For the section header of the very last tag (the first tag to be created in the repo), the compare link will now (instead of not existing) show a link to compare the last tag against the initial commit. You can see this behavior in the tests reflected by a `nil` for the older_tag value. 

Next, I don't think that `add_issues_wo_labels` and `add_pr_wo_labels` were actually implemented. When I ran the utility against changelog_test, it would still show unlabeled PRs and Issues in the changelog even when those two values were set to false. This has been addressed.

### CODE REVIEW

**Comment by thorsteneckel** [comment]
HI @tduffield - I like your commit. Actually we had the same issue in our company last week and I was fixing and debugging around trying to get the `add_issues_wo_labels` option working as expected. I found out that the `add_issues_wo_labels` option is only used when `include_labels` and/or `exclude_labels` are used, too.
Since I'm not deep enough into this I kept everything as it was and used a combination of these options. You approach is imho the right one. But since I'm not sure about it I'm really looking forward to a response of someone with deeper knowledge.

File: lib/github_changelog_generator/generator/generator.rb [hunk 1]
```diff
@@ -55,22 +55,31 @@ def encapsulate_string(string)
     # @param [Array] pull_requests List or PR's in new section
     # @param [Array] issues List of issues in new section
     # @param [String] newer_tag Name of the newer tag. Could be nil for `Unreleased` section
-    # @param [String] older_tag_name Older tag, used for the links. Could be nil for last tag.
+    # @param [String] older_tag Older tag, used for the links. Could be nil for last tag.
```
Comment (olleolleolle) [inline lib/github_changelog_generator/generator/generator.rb]: Doc block note: the older_tag is no longer a String, it's `[Hash,nil]`.

**Comment by olleolleolle** [comment]
Your added tests will support lots of refactoring, which was less possible before you came along. 

This is great. 

Thank you!

**Comment by wilson208** [comment]
I was experiencing issues with un-labeled PR's being included in the changelog and can confirm I have tested this PR and it addresses the issues I was seeing. Thanks a lot @tduffield!

Any idea of when/if this will be merged and released?

**Comment by tphoney** [comment]
Is there anything i can do, to help get this merged ?

**Comment by HAIL9000** [comment]
Hey again! Just wanted to check and see if there was anything I could do to help this PR along, we're currently blocked by this fix.

Thanks!

**Comment by hunner** [comment]
This can be closed; merged in #550

**Comment by olleolleolle** [comment]
Thanks for contributing this!

Closing this issue means progress was made, and you started that.

**Pull request closed without merging**